odoo/tools/pdf.py

```diff
@@ -19,3 +50,258 @@ def merge_pdf(pdf_data):
+        PDF/A standard.
+        Add an attachment to the pdf. The attachment will only be written in the file later in _embed_attachments.
+        :param fname: The name of the attached file
+        :param fdata: The data of the attached file
+        :param fsubtype: The eventual subtype of the attached file (required by PDF/A-3)
+        """
+        self._attachments_data.append({
+            'filename': fname,
+            'content': fdata,
+            'subtype': fsubtype,  # optional
+        })
+
+    def write(self, stream):
+        """
+        Override from the base class to support converting to PDF/A at the moment of writing.
+        :param stream: the stream, to which we are writing the output
+        """
+        if self._is_pdfa:
+            self._convert_to_pdfa(stream)
+        else:
+            self._embed_attachments()
+            super().write(stream)
+
+    def enable_pdfa(self):
+        """
+        Mark this writer as using pdfa. This will enable the conversion to pdf/a when writing to the stream.
+        """
+        self._is_pdfa = _get_ghostscript_bin() is not False
+
+    def _convert_to_pdfa(self, stream):
+        """
+        Take the content of a pdf generated by wkhtmltopdf and use ghostscript to make PDF/A-3B compliant.
+        :return: a PDF/A-3B compliant file
+        """
+        # Replace the DOCINFO as wkhtmltopdf export one with utf-16be characters and ghostscript doesn't like it.
+        info = self.getObject(self._info)
+        info.update({
+            NameObject("/Creator"): createStringObject(""),
+            NameObject("/Title"): createStringObject("")
+        })
+        super().write(stream)
+        pdfa_content = pdf_content = stream.getvalue()
+
+        gs_bin = _get_ghostscript_bin()
+        if gs_bin:
+            try:
+                cmd = [
+                    gs_bin,
+                    '-dQUIET',  # Remove all routine information from the output
+                    '-dPDFA=3',  # Generate a PDF/A-3 file
+                    '-dBATCH',
+                    '-dNOPAUSE',
+                    '-dSAFER',  # Using DSAFER disallow access to the files on the system.
+                    '-sColorConversionStrategy=UseDeviceIndependentColor',
+                    '-sProcessColorModel=DeviceRGB',
+                    '-sDEVICE=pdfwrite',
+                    '-dPDFACompatibilityPolicy=1',  # Discard features that would result in a not PDF/A compliant file
+                    '-sOutputFile=-',  # "-" as output file makes it write to stdout instead of a file
+                    '-',  # "-" as input file is required since we provide the data from stdin
+                ]
+                # Run the command in a new subprocess. Use communicate to send the file input, and get the output
+                process = Popen(cmd, stdin=PIPE, stderr=PIPE, stdout=PIPE)
+                pdfa_content, error = process.communicate(pdf_content)
```

Comment:
Since you're not actually using the streaming capabilities of Popen, why not just use `subprocess.run`?